File: spark/src/main/scala/io/github/interestinglab/waterdrop/spark/structuredstream/source/StructuredStreamingSource.scala
package io.github.interestinglab.waterdrop.spark.structuredstream.source

import io.github.interestinglab.waterdrop.apis.BaseSource
import io.github.interestinglab.waterdrop.spark.structuredstream.StructuredStreamingEnv
import org.apache.spark.sql.{Dataset, Row}

trait StructuredStreamingSource extends BaseSource[Dataset[Row], StructuredStreamingEnv]{

}

File: seatunnel-apis/seatunnel-api-spark/src/main/java/org/apache/seatunnel/spark/structuredstream/StructuredStreamingSource.java
package org.apache.seatunnel.spark.structuredstream;

import org.apache.seatunnel.spark.BaseSparkSource;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

@SuppressWarnings("PMD.AbstractClassShouldStartWithAbstractNamingRule")
public abstract class StructuredStreamingSource extends BaseSparkSource<Dataset<Row>> {
}
